content/en/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,207 @@
+
+A special thanks goes to the huge group of individuals involved in getting this past the finish line.
+
+## Trying it out
+
+While Kubernetes 1.28 is not out yet, and the feature is Alpha (and therefore, off by default), and the support for the feature in Istio is not yet shipped, we can still try it out today - just don't try this in prod!
+
+First, we need to spin up a pre-release Kubernetes cluster, with the `SidecarContainers` feature enabled:
+
+{{< text shell >}}
+$ cat <<EOF | kind create cluster --name sidecars --image gcr.io/istio-testing/kind-node:v1.28.0-beta.0 --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+featureGates:
+  SidecarContainers: true
+EOF
+{{< /text >}}
+
+Then we can download the latest Istio pre-release. I used Linux here; see [Dev Builds](https://github.com/istio/istio/wiki/dev-builds) for more steps.
+These images are built from the latest Istio commit, so again - do not try in prod!
+When we install, we will enable the feature flag for native sidecar support and turn on access logs to help demo things later.
+
+{{< text shell >}}
+$ TAG=$(curl https://storage.googleapis.com/istio-build/dev/latest)
+$ curl -L https://storage.googleapis.com/istio-build/dev/$TAG/istioctl-$TAG-linux-amd64.tar.gz | tar xz
+$ ./istioctl install --set values.pilot.env.ENABLE_NATIVE_SIDECARS=true -y --set meshConfig.accessLogFile=/dev/stdout
+{{< /text >}}
+
+And finally we can deploy a workload:
+
+{{< text shell >}}
+$ kubectl label namespace default istio-injection=enabled
+$ kubectl apply -f samples/sleep/sleep.yaml
+{{< /text >}}
+
+Let's look at the pod:
+
+{{< text shell >}}
+$ kubectl get pods
+NAME                     READY   STATUS    RESTARTS   AGE
+sleep-7656cf8794-8fhdk   2/2     Running   0          51s
+{{< /text >}}
+
+Everything looks normal at first glance...
+If we look under the hood, we can see the magic, though.
+
+{{< text shell >}}
+$ kubectl get pod -o "custom-columns="\
+"NAME:.metadata.name,"\
+"INIT:.spec.initContainers[*].name,"\
+"CONTAINERS:.spec.containers[*].name"
+
+NAME                     INIT                     CONTAINERS
+sleep-7656cf8794-8fhdk   istio-init,istio-proxy   sleep
+{{< /text >}}
+
+Here we can see all the `containers` and `initContainers` in the pod.
+
+Surprise! `istio-proxy` is now an `initContainer`.
+
+More specifically, it is an `initContainer` with `restartPolicy: Always` set (a new field, enabled by the `SidecarContainers` feature).
+This tells Kubernetes to treat it as a sidecar.
+
+This means that later containers in the list of `initContainers`, and all normal `containers` will not start until proxy container is first ready.
+Additionally, the pod will terminate even if the proxy container is still running.
+
+### Init container traffic
+
+To put this to the test, let's make our pod actually do something.
+Here we deploy a simple pod that sends a request in an `initContainer`.
+Normally, this would fail.
+
+{{< text yaml >}}
```

Comment:
Can you use the deploy command and mention this will have istio-proxy sidecar injected?  I assume this is to the default ns that has injection label.